src/plugins/formulas/formulas.js

```diff
@@ -177,7 +181,20 @@ export class Formulas extends BasePlugin {
       this.switchSheet(pluginSettings.sheetName);
     }
 
-    super.updatePlugin();
+    // If no data was passed to the `updateSettings` method and no sheet is connected to the instance -> create a
+    // new sheet using the currently used data. Otherwise, it will be handled by the `afterLoadData` call.
+    if (!newSettings.data && this.sheetName === null) {
```

Comment:
This one's pretty tricky. 
When calling `updateSettings` with `data` present, it calls `loadData`, when calling it without `data` - it doesn't. 
Now - we have the sheet-creation and data-filling logic in the `afterLoadData` hook callback (it has to be there, because of some data-modifying logic in `loadData`). This creates problems with no sheet being present in this issue's scenario, when the data is initialized first, and formulas are being enabled using `updateSettings` (without `data` provided).
Basically, this condition ensures the sheet creation logic is not doubled (`updatePlugin` + `onAfterLoadData`).

Reply:
Thanks!